src/ProcedureMigration.php

```diff
@@ -0,0 +1,211 @@
+     */
+    private bool $deterministic;
+
+    /**
+     * @var string
+     */
+    private string $readsSqlData;
+
+    /**
+     * @var ParameterOperation[]
+     */
+    private array $inParameterOperations;
+
+    /**
+     * @var ParameterOperation[]
+     */
+    private array $outParameterOperations;
+
+    /**
+     * @var string|null
+     */
+    private ?string $body = null;
+
+    /**
+     * Returns a new create procedure migration.
+     *
+     * @param string $name
+     * @return static
+     */
+    public static function create(string $name)
+    {
+        return new self($name, ProcedureOperation::CREATE);
+    }
+
+    /**
+     * Returns a new drop procedure migration.
+     *
+     * @param string $name
+     * @return static
+     */
+    public static function drop(string $name)
+    {
+        return new self($name, ProcedureOperation::DROP);
+    }
+
+    /**
+     * Migration constructor.
+     *
+     * @param string      $name
+     * @param string      $operation
+     * @param bool        $deterministic
+     * @param string      $readsSqlData
+     * @param array       $inParameterOperations
+     * @param array       $outParameterOperations
+     * @param string|null $body
+     */
+    private function __construct(
+        string $name,
+        string $operation,
+        bool $deterministic = false,
+        string $readsSqlData = 'READS SQL DATA',
+        array $inParameterOperations = [],
+        array $outParameterOperations = [],
+        string $body = null
+    ) {
+        $this->name = $name;
+        $this->operation = $operation;
+        $this->deterministic = $deterministic;
+        $this->readsSqlData = $readsSqlData;
+        $this->inParameterOperations = $inParameterOperations;
+        $this->outParameterOperations = $outParameterOperations;
+        $this->body = $body;
+    }
+
+    /**
+     * Pushes a new add column operation.
+     *
+     * @param string $body
+     * @return ProcedureMigration
+     */
+    public function withBody(string $body): ProcedureMigration
+    {
+        if ($this->operation !== ProcedureOperation::CREATE) {
+            throw new LogicException('Procedure body can only be set in a procedure create migration.');
+        }
+
+        $this->body = $body;
+
+        return $this;
+    }
+
+    /**
+     * Returns the table operation.
+     *
+     * @return ProcedureOperation
+     */
+    public function getOperation(): ProcedureOperation
+    {
+        return new ProcedureOperation(
+            $this->name,
+            $this->operation,
+            $this->deterministic,
+            $this->readsSqlData,
+            $this->inParameterOperations,
+            $this->outParameterOperations,
+            $this->body
+        );
+    }
+
+    /**
+     * Sets determinism for the function.
+     *
+     * @param bool $deterministic
+     * @return $this
+     */
+    public function isDeterministic(bool $deterministic): ProcedureMigration
```

Comment:
In keeping with the other factory methods, should this be called `withDeterminism`? The `is` prefix makes me think this is some form of getter without having a closer look.